File: common/frontend_theme.go
package common

import (
	"strings"

	"github.com/gin-gonic/gin"
)

const FrontendThemeCookieName = "frontend_theme"
const FrontendThemeCookieMaxAge = 60 * 60 * 24 * 365

func NormalizeFrontendTheme(theme string) string {
	theme = strings.ToLower(strings.TrimSpace(theme))
	if theme == "default" || theme == "classic" {
		return theme
	}
	return ""
}

func SetFrontendThemeCookie(c *gin.Context, theme string) {
	theme = NormalizeFrontendTheme(theme)
	if theme == "" {
		return
	}
	c.SetCookie(FrontendThemeCookieName, theme, FrontendThemeCookieMaxAge, "/", "", false, false)
}

File: web/default/src/lib/frontend-theme.ts
import { getCookie, setCookie } from '@/lib/cookies'

export const FRONTEND_THEME_COOKIE_NAME = 'frontend_theme'
export const FRONTEND_THEME_COOKIE_MAX_AGE = 60 * 60 * 24 * 365

export type FrontendTheme = 'default' | 'classic'

export function normalizeFrontendTheme(
  value?: string | null
): FrontendTheme {
  if (value === 'classic') {
    return 'classic'
  }
  return 'default'
}

export function getFrontendTheme(): FrontendTheme {
  return normalizeFrontendTheme(getCookie(FRONTEND_THEME_COOKIE_NAME))
}

export function setFrontendTheme(theme: FrontendTheme): void {
  setCookie(FRONTEND_THEME_COOKIE_NAME, theme, FRONTEND_THEME_COOKIE_MAX_AGE)
}

export function getFrontendThemeSettingsPath(theme: FrontendTheme): string {
  return theme === 'classic' ? '/console/personal' : '/profile'
}
